content/en/docs/Reference/declarative-config.md

```diff
@@ -0,0 +1,341 @@
+```cue
+#Package: {
+  schema: "olm.package"
+  
+  // Package name
+  name: string & !=""
+
+  // A description of the package
+  description?: string
+
+  // The package's default channel
+  defaultChannel: string & !=""
+  
+  // An optional icon
+  icon?: {
+    base64data: string
+    mediatype:  string
+  }
+}
+```
+
+#### `olm.bundle`
+
+<!--
+TODO(joelanford): Add a description of the `olm.bundle` schema here 
+-->
+
+The `olm.bundle` cue schema is:
+```cue
+#Bundle: {
+  schema: "olm.bundle"
+  package: string & !=""
+  name: string & !=""
+  image: string & !=""
+  properties: [...#Property]
+  relatedImages?: [...#RelatedImage]
+}
+
+#Property: {
+  // type is required
+  type: string & !=""
+
+  // value is required, and it must not be null
+  value: !=null
+}
+
+#RelatedImage: {
+  // image is the image reference
+  image: string & !=""
+  
+  // name is an optional descriptive name for an image that
+  // helps identify its purpose in the context of the bundle
+  name?: string & !=""
+}
+```
+
+### Properties
+
+Properties are arbitrary pieces of metadata that can be attached to DC schemas. The type field is a string that effectively specifies the semantic and syntactic meaning of the value field. The value can be any arbitrary JSON/YAML.
+
+
+OLM defines a handful of property types, again using the reserved `olm.*` prefix.
+
+#### `olm.package`
+
+An `olm.package` property defines the package name and version. This is a required property on bundles, and there must be exactly one of these properties. The `packageName` must match the bundle's first-class `package` field, and the `version` must be a valid [semantic version][semver]
+
+The `olm.package` property [cue][cuelang-spec] schema is:
+```cue
+#PropertyPackage: {
+  type: "olm.package"
+  value: {
+    packageName: string & !=""
+    version: string & !=""
+  }
+}
+```
+
+#### `olm.gvk`
+
+An `olm.gvk` property defines the group, version, and kind (GVK) of a Kubernetes API that is provided by this bundle. This property is used by OLM to resolve a bundle with this property as a dependency for other bundles that list the same GVK as a required API. The GVK must adhere to Kubernetes GVK validations.
+
+The `olm.gvk` property [cue][cuelang-spec] schema is:
+```cue
+#PropertyGVK: {
+  type: "olm.gvk"
+  value: {
+    group: string & !=""
+    version: string & !=""
+    kind: string & !=""
+  }
+}
+```
+
+#### `olm.channel`
+
+An `olm.channel` property defines a channel that a bundle is in, and optionally, the name of another bundle that it replaces in that channel.
+
+A bundle can include multiple `olm.channel` properties, but it is invalid to define multiple `olm.channel` properties for the same channel name.
+
+Lastly, it is valid for an `olm.channel`'s replaces value to reference another bundle that cannot be found in this index (or even another index) as long as other channel invariants still hold (e.g. a channel cannot have multiple heads).
```

Comment:
is is possible or preferable to allow any sort of metadata to be associated with a particular channel?  for example, an operand version or product version number.  In this way, the channel name does not have to be very complex to cover various scenarios, but instead channel metadata could be defined to create very complex channel definitions.

Reply:
This will likely be discussed in more detail in follow-on design discussions around upgrade graph APIs and a potential new `olm.channel` schema/blob.

So currently not possible, but I think probably the direction that we're headed.